jose/middleware.go

```
@@ -97,3 +99,95 @@ func (rt RoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 	}
 	return rt.RoundTripper.RoundTrip(r)
 }
+
+// authMetrics is a bundle of prometheus HTTP metrics recorders
+type authMetrics struct {
+	authSuccessCounter *prometheus.CounterVec
+	authFailureCounter *prometheus.CounterVec
+}
+
+// newAuthMetrics creates and returns a metrics bundle. The user may optionally
+// specify an existing Prometheus Registry. If no Registry is provided, the global Prometheus
+// Registry is used. Finally, if mustRegister is true, and a registration error is encountered,
+// the application will panic.
+func newAuthMetrics(registry prometheus.Registerer) authMetrics {
+	labels := []string{"path", "method"}
+	authSuccessCounter := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_authentication_success_total",
+			Help: "Total number of HTTP requests in which authentication succeeded",
+		},
+		labels,
+	)
+	authFailureCounter := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_authentication_failure_total",
+			Help: "Total number of HTTP requests in which authentication failed",
+		},
+		labels,
+	)
+	// If the user hasnt provided a Prometheus Registry, use the global Registry
+	if registry == nil {
+		registry = prometheus.DefaultRegisterer
+	}
+
+	toRegister := map[string]prometheus.Collector{
+		"authSuccessCounter": authSuccessCounter,
+		"authFailureCounter": authFailureCounter,
+	}
+	for name, collector := range toRegister {
+		if err := registry.Register(collector); err != nil {
```

I recently changed this in tools to just simply be `_ = registry.Register(...)` and ignore the error. If the user really wants an error they should indicate that they want it to register via `_ = registry.MustRegister(...)`